hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -62,15 +61,14 @@ public class HoodieLogFormatWriter implements HoodieLogFormat.Writer {
       Short replication,
       Long sizeThreshold,
       String rolloverLogWriteToken,
-      LogFileCreationCallback fileCreationHook) {
+      LogFileCreationCallback fileCreationCallback) {
```

Comment:
I see `HoodieWriteHandle#createLogWriter` create log writer without try with resources.  Will that be a problem without the shutdown hook?

Reply:
The writer is hold and maintained by the outer write handle, the write handle has its own lifecycle management, I think we should also fix that. But generally the write handle does not really implement `Closable`, we may need to enhance it later on.

Reply:
Hi, @bvaradar @nbalajee can you help to confirm whether it is safe to remove this shutdown hook from the log format writer?

Reply:
Looking at other places like HoodieWriteHandle#createLogWriter, we are passing the LogFormatWriter to the caller. So, it does not look like we can get away by try-with-resource refactoring alone.

Reply:
that's true, the shutdown hook is generally not a good way for releasing resource purpose, I'm just wondering the background for introducing it in the first place, what is the effect if the shutdown hook is not there and the inputstream is not closed ?

Reply:
I am not able to remember the specific reason. @n3nash @nbalajee : Any ideas ?

Reply:
why not we consider make HoodieIOHandle implement Closeable and enforce it everywhere? and so write handle does close the log format writer as well.

Reply:
> that's true, the shutdown hook is generally not a good way for releasing resource purpose, I'm just wondering the background for introducing it in the first place, what is the effect if the shutdown hook is not there and the inputstream is not closed ?

HoodieLogWriter has the mechanism to rollover to the next version of the log file and start writing to a new log file.   However, this might leave a trail of stale/partially written log files (incomplete or corrupt log blocks).   Improvements added for handling corrupt log blocks should be able to ignore these partial log blocks and recover automatically. 

However, from debugging the issues perspective, having too many partially written log files (log blocks) could be challenging to handle and mask some real issues.